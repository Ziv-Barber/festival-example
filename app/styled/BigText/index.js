// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Material-ui stuff:
import MuiTypography from '@material-ui/core/Typography'

// Our types:
import type { BigTextPropsType } from './index.types'

/**
 * BigText react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const BigText = (props: BigTextPropsType) => (
  <MuiTypography
    className={props.className || 'bigText'}
    variant="h5"
    component="h2"
  >
    {props.children}
  </MuiTypography>
)

export default styled(BigText)`
  position: relative;
  margin: ${(props) => props.theme.spacing.unit}px !important;
  padding: ${(props) => props.theme.spacing.unit * 2}px !important;
  border-radius: ${(props) => props.theme.spacing.unit * 2}px !important;
  background: ${(props) => props.theme.palette.primary.main};
  color: #fff;

  & ::after {
    content: '';
    position: absolute;
    top: -96px;
    right: 0;
    width: 0;
    height: 0;
    border-bottom: 96px solid
      ${(props) => props.theme.palette.background.default};
    border-right: 100vw solid transparent;
    z-index: 1;
  }
`

// In case that someone need our types:
export type { BigTextPropsType as Props }
