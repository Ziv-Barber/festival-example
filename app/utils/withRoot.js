// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
