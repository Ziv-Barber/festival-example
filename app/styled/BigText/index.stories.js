import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import BigText from './index'

storiesOf('BigText', module).add(
  'basic',
  () => <BigText text="Basic example" />,
  { notes: { markdown: 'This is an example to the BigText component.' } }
)
