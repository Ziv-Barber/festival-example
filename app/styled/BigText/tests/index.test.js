// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import BigText from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {BigText} */
describe('<BigText />', () => {
  afterEach(cleanup)

  /** @test {BigText} */
  it('should match snapshort', () => {
    const CompToTest = () => <BigText>This is a text</BigText>
    const App = withRoot(CompToTest)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  /** @test {BigText} */
  it('should render', () => {
    const CompToTest = () => <BigText>This is a text</BigText>
    const App = withRoot(CompToTest)
    const { getByText } = render(<App />)
    expect(getByText(/^This is a text/)).not.toBeNull()
  })
})
