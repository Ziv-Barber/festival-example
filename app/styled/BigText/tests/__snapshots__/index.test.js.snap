// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigText /> should match snapshort 1`] = `
<div>
  <h2
    class="MuiTypography-root-1 MuiTypography-h5-17 BigText-sc-1k80hov-0 gwRgJq"
  >
    This is a text
  </h2>
</div>
`;
